resources/qml/ui/TimelineEffects.qml

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import QtQuick 2.15
+import QtQuick.Particles 2.15
+
+Item {
+    readonly property int maxLifespan: Math.max(confettiEmitter.lifeSpan, rainfallEmitter.lifeSpan)
+
+    function pulseConfetti()
+    {
+        confettiEmitter.pulse(parent.height * 2)
+    }
+
+    function pulseRainfall()
+    {
+        rainfallEmitter.pulse(parent.height * 7.5)
```

Comment:
I don't think this should be dependent on the window size?

Reply:
Basically emit for a specific time, then let them just fall down, instead of emitting for longer, when the window is bigger?

Reply:
The idea is to emit an appropriate number of particles to make sure that different window sizes have roughly the same particle density. I guess there comes a point where you just don't bother covering the entire screen though.

Reply:
Mhm, currently the rain just lasts a bit too long imo.

Reply:
I had thought about that too. I'll try to scale it down a bit.